tensorflow_addons/losses/giou_loss.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements GIoU loss."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+class GIoULoss(tf.keras.losses.Loss):
+    """Implements the GIoU loss function.
+
+    GIoU loss was first introduced in the Generalized Intersection over Union
+    paper (https://giou.stanford.edu/GIoU.pdf). GIoU is a enhance for model
```

Comment:
Use `[Generalized Intersection over Union paper](https://giou.stanford.edu/GIoU.pdf)` for hyperlink while generating docs.